src/OculusHandModel.ts

```diff
@@ -0,0 +1,112 @@
+import { Object3D, Sphere, Box3, Mesh, Texture, Vector3, EventListener, Event } from 'three'
+import { XRHandMeshModel } from './XRHandMeshModel'
+const TOUCH_RADIUS = 0.01
+const POINTING_JOINT = 'index-finger-tip'
+
+export interface XRButton extends Object3D {
+  onPress(): void
+  onClear(): void
+  isPressed(): boolean
+  whilePressed(): void
+}
+
+class OculusHandModel extends Object3D {
+  controller: Object3D
+  motionController: XRHandMeshModel | null
+  envMap: Texture | null
+  mesh: Mesh | null
+  xrInputSource: XRInputSource | null
+
+  leftModelPath?: string
+  rightModelPath?: string
+
+  constructor(controller: Object3D, leftModelPath?: string, rightModelPath?: string) {
+    super()
+
+    this.controller = controller
+    this.motionController = null
+    this.envMap = null
+    this.leftModelPath = leftModelPath
+    this.rightModelPath = rightModelPath
+
+    this.mesh = null
+    this.xrInputSource = null
+
+    controller.addEventListener('connected', this._onConnected)
+    controller.addEventListener('disconnected', this._onDisconnected)
+  }
+
+  private _onConnected: EventListener<Event, 'connected', Object3D<Event>> = (event) => {
+    const xrInputSource = event.data
+
+    if (xrInputSource.hand && !this.motionController) {
+      this.xrInputSource = xrInputSource
+
+      this.motionController = new XRHandMeshModel(
+        this,
+        this.controller,
+        undefined,
+        xrInputSource.handedness,
+        xrInputSource.handedness === 'left' ? this.leftModelPath : this.rightModelPath
+      )
+    }
+  }
+
+  private _onDisconnected: EventListener<Event, 'disconnected', Object3D<Event>> = () => {
```

Comment:
**suggestion:** lets's also add `if (xrInputSource.hand` guard to not to accidently clear it if some non hand-related disconnect event creeps in